packages/app-builder/src/routes/_builder+/settings+/api-keys.tsx

```diff
@@ -1,3 +1,135 @@
+import {
+  Callout,
+  CollapsiblePaper,
+  CopyToClipboardButton,
+  Page,
+} from '@app-builder/components';
+import { isAdmin } from '@app-builder/models';
+import { type ApiKey, type CreatedApiKey } from '@app-builder/models/api-keys';
+import { CreateApiKey } from '@app-builder/routes/ressources+/settings+/api-keys+/create';
+import { DeleteApiKey } from '@app-builder/routes/ressources+/settings+/api-keys+/delete';
+import { tKeyForApiKeyRole } from '@app-builder/services/i18n/translation-keys/api-key';
+import { serverServices } from '@app-builder/services/init.server';
+import { getRoute } from '@app-builder/utils/routes';
+import { json, type LoaderFunctionArgs, redirect } from '@remix-run/node';
+import { useLoaderData } from '@remix-run/react';
+import { createColumnHelper, getCoreRowModel } from '@tanstack/react-table';
+import clsx from 'clsx';
+import { useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Table, useTable } from 'ui-design-system';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const { authService, authSessionService } = serverServices;
+  const { apiKey, user } = await authService.isAuthenticated(request, {
+    failureRedirect: getRoute('/sign-in'),
+  });
+  if (!isAdmin(user)) {
+    return redirect(getRoute('/'));
+  }
+  const apiKeys = await apiKey.listApiKeys();
+
+  const authSession = await authSessionService.getSession(request);
+  const createdApiKey = authSession.get('createdApiKey');
+  const headers = new Headers();
+  if (createdApiKey) {
+    headers.set(
+      'Set-Cookie',
+      await authSessionService.commitSession(authSession),
+    );
+  }
+
+  return json(
+    { apiKeys, createdApiKey },
+    {
+      headers,
+    },
+  );
+}
+
+const columnHelper = createColumnHelper<ApiKey>();
+
 export default function ApiKeys() {
-  return <div>ApiKeys</div>;
```

Comment:
I'm asking because I couldn't find it: did you put a check on the authenticated user role somewhere before displaying this ? Is it somewhere "higher up" in the file hierarchy ? Because effectively this should only be available for admins

Reply:
It's "higher up" in the file, look at the `loader` :
```ts
const { apiKey, user } = await authService.isAuthenticated(request, {
    failureRedirect: getRoute('/sign-in'),
  });
  if (!isAdmin(user)) {
    return redirect(getRoute('/'));
  }
```